src/components/user/tour/create-tour/guests.js

```diff
@@ -0,0 +1,204 @@
+import React, {Component} from "react";
+import Button from "@material-ui/core/Button";
+import {DataGrid} from "@material-ui/data-grid";
+import TextField from "@material-ui/core/TextField";
+import Checkbox from "@material-ui/core/Checkbox";
+import CheckBoxOutlineBlankIcon from '@material-ui/icons/CheckBoxOutlineBlank';
+import CheckBoxIcon from '@material-ui/icons/CheckBox';
+import {GUESTS_COLUMNS, GUESTS_ROWS} from "../utils/constants";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+
+export default class Guest extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            guestsGridColumns: GUESTS_COLUMNS,
+            searchValue: ''
+        };
+        this.handleSaveClick = this.handleSaveClick.bind(this);
+        this.handleInputChange = this.handleInputChange.bind(this)
+    }
+
+    handleSaveClick() {
+        const {formValues, handleAddData, currentStepDataKey} = this.props;
+        handleAddData(currentStepDataKey, formValues)
+    }
+
+    handleInputChange(event) {
+        const target = event.target;
+        const value = target.type === 'checkbox' ? target.checked : target.value;
+        const name = target.name;
+
+        this.setState({
+            [name]: value
+        });
+    }
+
+    render() {
+        const {currentStepName, handleInputChange, formValues, data} = this.props;
+        const {guestsGridColumns, searchValue} = this.state;
+        const searchedDataRows = data.filter(item => (item.firstName + ' ' + item.lastName).toLowerCase().includes(searchValue.toLowerCase()));
```

Comment:
Search functionality must be out of render function, otherwise, calculations will be done after each update of the component.